subsys/secure_storage/src/aead/aead_key_hash.c

```diff
@@ -0,0 +1,23 @@
+/*
+ * Copyright (c) 2023 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#include <psa/crypto.h>
+#include <stdio.h>
+
+#include "aead_key.h"
+
+psa_status_t secure_storage_get_key(psa_storage_uid_t uid, uint8_t *key_buf, size_t key_length)
+{
+
+	size_t olen;
+
+	if (key_length < AEAD_KEY_SIZE) {
+		return PSA_ERROR_BUFFER_TOO_SMALL;
+	}
+
+	return psa_hash_compute(PSA_ALG_SHA_256, (uint8_t *)&uid, sizeof(uid), key_buf, key_length,
```

Comment:
It seems odd to me to get the key this way. Isn't HKDF key derivation better?

Reply:
Hmm, settings storage is sometimes used to store keys, but that defeats the purpose of storing things encrypted in settings. Unless you just want to get this part working, however we didn't add an "unencrypted backend" so I guess we need this as our just-works solution?

I think giving this option a help text of "this is unsecure" would be good, to make sure nobody uses this by accident and thinks it provides them any good security.

Reply:
OK Added and unsecure and not recommended block in the help text